### ISSUE TITLE
$Env:AWS_PROFILE disregarded when set to a profile defined in ~/.aws/config ("No valid credential sources...")

### ISSUE THREAD

**Issue by bgshacklett** [terraform-providers/terraform-provider-aws#10066]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.8
+ provider.aws v2.27.0

### Affected Resource(s)

N/A—Provider will not initialize

### Environment
```
> gi Env:/AWS_PROFILE

Name                           Value
----                           -----
AWS_PROFILE                    k8s-lab
```

### Terraform Configuration Files

```hcl
# provider.tf
provider "aws" {
  version = "2.27.0"
  region  = "us-east-2"
}

# terraform.tf
terraform {
  required_version  = ">=0.12.0"
}

# main.tf
data "aws_region" "current" {}
```

### Debug Output

https://gist.github.com/bgshacklett/3f205d2540dd9a42ae540cf7192a11f7

### Expected Behavior
Terraform successfully generates a plan

### Actual Behavior
An error is generated:
```
Error: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
```

### Steps to Reproduce

1. Configure a profile which assumes a role in `~/.aws/config`:
    ```
    [profile k8s-lab]
    source_profile = Lab-LabUser
    role_arn = arn:aws:iam::[redacted]:role/Admin
    ```
1. `terraform plan`

### Important Factoids

* I'm able to run `aws sts get-caller-identity` successfully:
    ```
    > aws sts get-caller-identity
    {
        "UserId": "[redacted]:botocore-session-[redacted]",
        "Account": "[redacted]",
        "Arn": "arn:aws:sts::[redacted]:assumed-role/Admin/botocore-session-[redacted]"
    }
    ```
* Assigning `$Env:AWS_PROFILE` to a profile configured in `.aws/credentials` (`Lab-LabUser` in this case) allows me to run a plan successfully.
* Explicitly configuring the role `k8s-lab` in the provider allows me to run a plan successfully.

### References

n/a

**Comment by aeschright** [comment]
Hi @bgshacklett ! Thanks for reporting this. Issues around assuming a role through the config profile settings will be resolved with #10379 which is scheduled for the next release.

**Comment by bflad** [comment]
Closing as #10379 was merged previously and v2.32.0 has been released. 👍
